azure-pipelines-pr.yml

```diff
@@ -0,0 +1,166 @@
+# Spark .NET build
+
+trigger:
+  batch: true
+  branches:
+    include:
+    - main
+
+variables:
+  buildConfiguration: 'Release'
+  _SignType: real
+  _TeamName: DotNetSpark
+  MSBUILDSINGLELOADCONTEXT: 1
+  ArtifactPath: '$(Build.ArtifactStagingDirectory)\Microsoft.Spark.Binaries'
+
+  backwardCompatibleRelease: '2.0.0'
+  forwardCompatibleRelease: '2.0.0'
+
+  backwardCompatibleTestOptions_Windows_2_4: ""
```

Comment:
Do we need them?

Reply:
These variables bring from original yaml file, they are used if backwardCompatibleTest and forwardCompatibleTest set to true.

Reply:
Do you know where they are used? I didn't find.

Reply:
These variables are referenced by generating variable name dynamically according to pool type and spark version, code as below:
```
backwardCompatibleTestOptions: $(backwardCompatibleTestOptions_${{ pool }}_${{ split(version, '.')[0] }}_${{ split(version, '.')[1] }})  
forwardCompatibleTestOptions: $(forwardCompatibleTestOptions_${{ pool }}_${{ split(version, '.')[0] }}_${{ split(version, '.')[1] }})
```
`backwardCompatibleTestOptions` and `forwardCompatibleTestOptions` used in azure-pipelines-e2e-tests-template.yml file.

Reply:
I see, thanks for explanation!